File: frontend/src/api/teachingMaterial.ts
/**
 * 高校教学就业平台 - 教学资料相关 API
 */
import request, { http } from '@/utils/request'

export interface TeachingMaterial {
  id?: number
  materialName: string
  materialType: string
  fileUrl: string
  fileSize?: number
  courseId?: number
  teacherId: number
  category?: string
  tags?: string
  description?: string
  isPublic?: number
  downloadCount?: number
  viewCount?: number
  status?: string
  createTime?: string
  updateTime?: string
  // 关联字段
  courseName?: string
  teacherName?: string
  fileExtension?: string
}

export interface TeachingMaterialQueryParams {
  current?: number
  size?: number
  teacherId?: number
  courseId?: number
  materialType?: string
  category?: string
  keyword?: string
}

export const teachingMaterialApi = {
  /**
   * 分页查询教学资料列表
   */
  getPage: (params: TeachingMaterialQueryParams) => {
    return http.get<any>('/teaching-material/page', { params })
  },

  /**
   * 根据ID获取教学资料详情
   */
  getById: (id: number) => {
    return http.get<TeachingMaterial>(`/teaching-material/${id}`)
  },

  /**
   * 上传教学资料
   */
  upload: (data: FormData) => {
    return http.post<any>('/teaching-material/upload', data)
  },

  /**
   * 更新教学资料信息
   */
  update: (data: TeachingMaterial) => {
    return http.put<boolean>('/teaching-material', data)
  },

  /**
   * 删除教学资料
   */
  delete: (id: number) => {
    return http.delete<boolean>(`/teaching-material/${id}`)
  },

  /**
   * 批量删除教学资料
   */
  batchDelete: (ids: number[]) => {
    return http.delete<boolean>('/teaching-material/batch', { data: ids })
  },

  /**
   * 记录下载
   */
  recordDownload: (id: number) => {
    return http.post<boolean>(`/teaching-material/${id}/download`)
  },

  /**
   * 记录查看
   */
  recordView: (id: number) => {
    return http.post<boolean>(`/teaching-material/${id}/view`)
  }
}

/**
 * 格式化文件大小
 */
export function formatFileSize(bytes: number): string {
  if (bytes === 0) return '0 B'
  const k = 1024
  const sizes = ['B', 'KB', 'MB', 'GB']
  const i = Math.floor(Math.log(bytes) / Math.log(k))
  return Math.round((bytes / Math.pow(k, i)) * 100) / 100 + ' ' + sizes[i]
}

/**
 * 获取资料类型图标
 */
export function getMaterialTypeIcon(type: string): string {
  const iconMap: Record<string, string> = {
    'ppt': '📊',
    'video': '🎥',
    'document': '📄',
    'image': '🖼️',
    'other': '📁'
  }
  return iconMap[type] || iconMap['other']
}

/**
 * 获取资料类型标签
 */
export function getMaterialTypeLabel(type: string): string {
  const labelMap: Record<string, string> = {
    'ppt': 'PPT课件',
    'video': '视频资料',
    'document': '文档资料',
    'image': '图片资料',
    'other': '其他资料'
  }
  return labelMap[type] || '其他资料'
}

File: backend/src/main/java/com/teaching/employment/entity/TeachingMaterial.java
package com.teaching.employment.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 教学资料实体类
 *
 * @author Teaching Employment Platform Team
 * @since 2026-01-17
 */
@Data
@TableName("t_teaching_material")
@ApiModel(description = "教学资料信息")
public class TeachingMaterial implements Serializable {

    private static final long serialVersionUID = 1L;

    @ApiModelProperty(value = "资料ID")
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @ApiModelProperty(value = "资料名称")
    private String title;

    @ApiModelProperty(value = "资料类型：ppt-video-document-image-other")
    private String materialType;

    @ApiModelProperty(value = "文件URL")
    private String fileUrl;

    @ApiModelProperty(value = "文件大小(字节)")
    private Long fileSize;

    @ApiModelProperty(value = "关联课程ID")
    private Long courseId;

    @ApiModelProperty(value = "教师ID")
    private Long teacherId;

    @ApiModelProperty(value = "分类标签")
    private String category;

    @ApiModelProperty(value = "标签(逗号分隔)")
    private String tags;

    @ApiModelProperty(value = "资料描述")
    private String description;

    @ApiModelProperty(value = "是否对学员公开：0-私有 1-公开")
    private Integer isPublic;

    @ApiModelProperty(value = "下载次数")
    private Integer downloadCount;

    @ApiModelProperty(value = "查看次数")
    private Integer viewCount;

    @ApiModelProperty(value = "状态：active-激活 inactive-停用")
    private String status;

    @ApiModelProperty(value = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @ApiModelProperty(value = "更新时间")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    // ==================== 关联查询字段（不映射到数据库） ====================

    @ApiModelProperty(value = "课程名称")
    @TableField(exist = false)
    private String courseName;

    @ApiModelProperty(value = "教师姓名")
    @TableField(exist = false)
    private String teacherName;

    @ApiModelProperty(value = "文件扩展名")
    @TableField(exist = false)
    private String fileExtension;
}
